peer.go

```diff
@@ -199,6 +201,9 @@ func (p *Peer) sendLoop() {
 	defer p.conn.Close() // close connection on fatal error
 	for {
 		select {
+		case <-p.net.stopper:
+			p.Stop()
+			return
```

Comment:
Shouldn't this logic be in its own go routine? Because it's not particularly related to Send logic, at least no more than read logic for instance. It's a bit cleaner and clearer if this go routine is exited through a single path (the `case <-p.stop`).

Reply:
I think that's a fair request from a theoretical standpoint but from a practical one, having another long-term goroutine in the system that effectively adds nothing isn't necessary. I think the performance impact is negligible enough that if you feel it's worth it, I'll add it.

Reply:
Nah it's fine.